networkx/generators/tests/test_classic.py

```diff
@@ -372,30 +411,57 @@ def test_path_graph(self):
         G = nx.path_graph("abc")
         assert len(G) == 3
         assert G.size() == 2
+        G = nx.path_graph("abcb")
+        assert len(G) == 3
+        assert G.size() == 2
         g = nx.path_graph("abc", nx.DiGraph)
         assert len(g) == 3
         assert g.size() == 2
         assert g.is_directed()
+        g = nx.path_graph("abcb", nx.DiGraph)
+        assert len(g) == 3
+        assert g.size() == 3
 
-    def test_star_graph(self):
-        star_graph = nx.star_graph
-        assert is_isomorphic(star_graph(0), nx.empty_graph(1))
-        assert is_isomorphic(star_graph(1), nx.path_graph(2))
-        assert is_isomorphic(star_graph(2), nx.path_graph(3))
-        assert is_isomorphic(star_graph(5), nx.complete_bipartite_graph(1, 5))
+        G = nx.path_graph((1, 2, 3, 2, 4))
+        assert G.has_edge(2, 4)
 
-        s = star_graph(10)
+    def test_star_graph(self):
+        assert is_isomorphic(nx.star_graph(""), nx.empty_graph(0))
+        assert is_isomorphic(nx.star_graph([]), nx.empty_graph(0))
+        assert is_isomorphic(nx.star_graph(0), nx.empty_graph(1))
+        assert is_isomorphic(nx.star_graph(1), nx.path_graph(2))
+        assert is_isomorphic(nx.star_graph(2), nx.path_graph(3))
+        assert is_isomorphic(nx.star_graph(5), nx.complete_bipartite_graph(1, 5))
+
+        s = nx.star_graph(10)
         assert sorted(d for n, d in s.degree()) == [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 10]
 
-        pytest.raises(nx.NetworkXError, star_graph, 10, create_using=nx.DiGraph)
+        pytest.raises(nx.NetworkXError, nx.star_graph, 10, create_using=nx.DiGraph)
 
-        ms = star_graph(10, create_using=nx.MultiGraph)
+        ms = nx.star_graph(10, create_using=nx.MultiGraph)
         assert edges_equal(ms.edges(), s.edges())
 
-        G = star_graph("abcdefg")
+        G = nx.star_graph("abc")
+        assert len(G) == 3
+        assert G.size() == 2
+
+        G = nx.star_graph("abcb")
+        assert len(G) == 3
+        assert G.size() == 2
+        G = nx.star_graph("abcb", create_using=nx.MultiGraph)
+        assert len(G) == 3
+        assert G.size() == 3
+
+        G = nx.star_graph("abcdefg")
         assert len(G) == 7
         assert G.size() == 6
 
+        # test non-int integers
+        np = pytest.importorskip("numpy")
+        G = nx.star_graph(np.int32(3))
+        assert len(G) == 4
+        assert G.size() == 3
```

Comment:
We probably want to break this out into it's own test, otherwise the entire `test_star_graph` test will be skipped if numpy isn't avialable.